app/src/main/java/com/pysiak/simple/clock/ui/clock/ClockFragment.java

```diff
@@ -24,7 +43,51 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
-        ClockPresenter clockPresenter = new ClockPresenter();
+        Mapbox.getInstance(getContext(), getString(R.string.map_box_key));
+        clockPresenter = new ClockPresenter(getContext());
         clockPresenter.attachView(this);
+        textView = view.findViewById(R.id.text_location_id);
+        view.findViewById(R.id.fab_location_search).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                PlaceAutocomplete.clearRecentHistory(getContext());
+                Intent intent = new PlaceAutocomplete.IntentBuilder()
+                        .accessToken(Mapbox.getAccessToken() != null ? Mapbox.getAccessToken() : getString(R.string.map_box_key))
+                        .placeOptions(PlaceOptions.builder()
+                                .backgroundColor(Color.parseColor("#EEEEEE"))
+                                .limit(LIMIT_PLACES)
+                                .build(PlaceOptions.MODE_CARDS))
+                        .build(getActivity());
+                startActivityForResult(intent, REQUEST_CODE_AUTOCOMPLETE);
+            }
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (resultCode == Activity.RESULT_OK && requestCode == REQUEST_CODE_AUTOCOMPLETE && data != null) {
+            CarmenFeature selectedCarmenFeature = PlaceAutocomplete.getPlace(data);
+            double lat = selectedCarmenFeature.center().latitude();
```

Comment:
Call for Api should be called on presenter side.
So you need to call here presenter method like requestTimeZone and add move all code under this line to this method.
On callback's methods (onResponse/onFailure) you need to call method from fragment like showResults and provide timezone result to it. Also don't forget to add method declaration to View